pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go

```diff
@@ -305,16 +326,20 @@ func syncClaim(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCli
 			return fmt.Errorf("Unexpected error getting persistent volume: %v\n", err)
 		}
 
+		// TODO:  Can this be removed? The ref will not be nil for volumes bound by this controller.
 		if volume.Spec.ClaimRef == nil {
 			glog.V(5).Infof("Rebuilding bind on pv.Spec.ClaimRef\n")
 			claimRef, err := api.GetReference(claim)
 			if err != nil {
 				return fmt.Errorf("Unexpected error getting claim reference: %v\n", err)
 			}
-			volume.Spec.ClaimRef = claimRef
-			_, err = binderClient.UpdatePersistentVolume(volume)
-			if err != nil {
+			clone, err := conversion.NewCloner().DeepCopy(volume)
+			volumeClone := clone.(*api.PersistentVolume)
+			volumeClone.Spec.ClaimRef = claimRef
+			if updatedVolume, err := binderClient.UpdatePersistentVolume(volumeClone); err != nil {
 				return fmt.Errorf("Unexpected error saving PersistentVolume.Status: %+v", err)
+			} else {
+				volumeIndex.Update(updatedVolume)
 			}
 		}
 
```

Comment:
I thought about that while leaving the TODO.  There is no api.ClaimError phase.  

I set phase back to Pending and returned an error.